### PR TITLE
test: add basic extend range protocol tests

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/427?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#427** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/427?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#426** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/426?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#424](https://github.com/thrumdev/nomt/pull/424) <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/424?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#422** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/422?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#421** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/421?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#420** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/420?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#419** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/419?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gabriele-0201 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/tests/extend_range_protocol.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
+mod common;
+use common::Test;
+use std::path::Path;
+
+#[rustfmt::skip]
+const KEYS_AND_VALUE_SIZES: [(u8, usize); 16] =[
```
Comment (rphmeier) [inline nomt/tests/extend_range_protocol.rs:6]: this seems like a good approach. but it is kind of fragile to the page size (though a lot of our code is). looks good, but add a comment explaining why this forces new leaves!

**Review by rphmeier — APPROVED** [review]

File: nomt/tests/extend_range_protocol.rs (hunk 1 above)
Comment (gabriele-0201) [reply nomt/tests/extend_range_protocol.rs:6]: Yes! Those numbers are tightly related to the size of a node/page! I will add a comment and the assertion!

Comment (gabriele-0201) [reply nomt/tests/extend_range_protocol.rs:6]: well, unfortunately both `BRANCH_NODE_SIZE` and `PAGE_SIZE` are inside private modules, not reachable from integration testing. I will add a clear comment explaining the dependency from the size of the node

Comment (rphmeier) [reply nomt/tests/extend_range_protocol.rs:15]: perfect

**Comment by rphmeier** [comment]
### Merge activity

* **Oct 8, 3:02 PM EDT**: @rphmeier started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/427).

**Pull request merged**